### PR TITLE
AB#3024 Create a way for Mocks to ingest a JSON file

### PR DESCRIPTION
**Pull request by Fbasham**
### Description
"PowerPlatform will send us a JSON file containing the data that would be returned by their API. If we could find a way to use these to feed the Mocks so that if there are any changes to the datasets, we can simply update the JSON files"

### Related Azure Boards Work Items
[AB#3024](https://dev.azure.com/DTS-STN/1fc40a8f-28cf-47bc-b6e4-1c234bd06177/_workitems/edit/3024)

### Checklist

- [x] I have tested the changes locally
- [ ] I have updated the documentation if necessary
- [ ] I have added/updated tests that prove my fix is effective or that my feature works
- [x] I have checked that my code follows the project's coding style by running `npm run format:check`
- [x] I have checked that my code contains no linting errors by running `npm run lint`
- [x] I have checked that my code contains no type errors by running `npm run typecheck`
- [x] I have checked that all unit tests pass by running `npm run test:unit -- run`
- [x] I have checked that all e2e tests pass by running `npm run test:e2e`

### CODE REVIEW

File: frontend/public/data/countries.json
Comment (gregory-j-baker) [inline frontend/public/data/countries.json:1]: I don't think this file should be in the `public/` folder. It should be in `app/` somewhere (possibly `app/mocks/`).

File: frontend/public/data/federalPrograms.json
Comment (gregory-j-baker) [inline frontend/public/data/federalPrograms.json:1]: I don't think this file should be in the `public/` folder. It should be in `app/` somewhere (possibly `app/mocks/`).

Comment (gregory-j-baker) [inline frontend/public/data/federalPrograms.json:1]: **Nitpick:** our file naming convention suggests that this file should be named `federal-programs.json`.

File: frontend/public/data/regions.json
Comment (gregory-j-baker) [inline frontend/public/data/regions.json:1]: I don't think this file should be in the `public/` folder. It should be in `app/` somewhere (possibly `app/mocks/`).

**Review by gregory-j-baker — CHANGES_REQUESTED** [review]

File: frontend/app/mocks/db.ts [hunk 1]
```diff
@@ -280,79 +261,32 @@ db.maritalStatus.create({
...
-  countryId: 'USA',
-  nameEnglish: 'New York',
-  nameFrench: 'État de New York',
-});
+countriesJson.value.forEach((country) =>
+  db.country.create({
+    countryId: country.esdc_countryid,
+    countryCode: country.esdc_countrycodealpha3,
+    nameEnglish: country.esdc_nameenglish,
+    nameFrench: country.esdc_namefrench,
```
Comment (gregory-j-baker) [inline frontend/app/mocks/db.ts:269]: The convention in this file is to use `nameEn` and `nameFr`. I think we should stick with that.

File: frontend/app/mocks/db.ts [hunk 2]
```diff
@@ -280,79 +261,32 @@ db.maritalStatus.create({
...
+  }),
+);
+
+// seed region list
+regionsJson.value.forEach((region) =>
+  db.region.create({
+    countryId: db.country.findFirst({ where: { countryId: { equals: region._esdc_countryid_value } } })?.countryCode,
+    provinceTerritoryStateId: region.esdc_internationalalphacode,
+    nameEnglish: region.esdc_nameenglish,
+    nameFrench: region.esdc_namefrench,
```
Comment (gregory-j-baker) [inline frontend/app/mocks/db.ts:279]: The convention in this file is to use `nameEn` and `nameFr`. I think we should stick with that.

**Review by gregory-j-baker — CHANGES_REQUESTED** [review]

File: frontend/public/data/federalPrograms.json
Comment (Fbasham) [reply frontend/public/data/federalPrograms.json:1]: changed

File: frontend/app/mocks/db.ts (hunk 1 above)
Comment (Fbasham) [reply frontend/app/mocks/db.ts]: updated

**Review by gregory-j-baker — DISMISSED** [review]

**Review by nicholas-ly — APPROVED** [review]

**Pull request merged**